packages/experimental/Avatar/SPEC.md

```diff
@@ -0,0 +1,311 @@
+   * Badge shows the avatar's presence status.
+   * Badge can be shown only if `active` prop is undefined or unset
+   */
+  badge?: PresenceBadgeProps;
+
+  /**
+   * Icon to be displayed when the avatar doesn't have an image or initials.
+   *
+   * @defaultvalue `PersonRegular` (the default icon's size depends on the Avatar's size)
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * Specify a string to be used instead of the name, to determine which color to use when color="colorful".
+   * Use this when a name is not available, but there is another unique identifier that can be used instead.
+   */
+  idForColor?: string | undefined;
+
+  /**
+   * The Avatar's image.
+   */
+  image?: ImageProps;
+
+  /**
+   * (optional) Custom initials.
+   *
+   * It is usually not necessary to specify custom initials; by default they will be derived from the `name` prop,
+   * using the `getInitials` function.
+   *
+   * The initials are displayed when there is no image (including while the image is loading).
+   */
+  initials?: string;
+
+  /**
+   * The name of the person or entity represented by this Avatar. This should always be provided if it is available.
+   *
+   * The name will be used to determine the initials displayed when there is no icon, as well as provided to
+   * accessibility tools.
+   */
+  name?: string;
+
+  /**
+   * Size of the avatar in pixels.
+   *
+   * Size is restricted to a limited set of supported values recommended for most uses (see `AvatarSizeValue`) and
+   * based on design guidelines for the Avatar control.
+   *
+   * @defaultvalue 24
+   */
+  size?: AvatarSize;
+
+  /**
+   * The avatar can have a circular or square shape.
+   * @defaultvalue circular
+   */
+  shape?: AvatarShape;
+
+  /**
+   * Currently used for image URL, will be deprecated when image prop works
+   */
+  src?: string;
+}
+```
+
+### Styling Tokens
+
+Tokens can be used to customize the styling of the control by using the `customize` function on the `Avatar`. For more information on using the `customize` API, please see [this page](../../framework/composition/README.md). The `Avatar` has the following tokens:
+
+```ts
+export interface JSAvatarTokens extends IBackgroundColorTokens, IForegroundColorTokens, AvatarConfigurableProps, IBorderTokens, FontTokens {
+  /**
+   * Avatar ocality which is changed depending on `active` prop.
+   */
+  avatarOpacity?: number;
+
+  /**
+   * The size of presence badge.
+   */
+  badgeSize?: BadgeSize;
+  /**
+   * The icon color.
+   */
+  iconColor?: ColorValue;
+
+  /**
+   * The size of the icon.
+   */
+  iconSize?: number;
+  /**
+   * The height of the avatar.
+   */
+  height?: number;
+
+  /**
+   * Avatar shapes:
+   */
+  circular?: JSAvatarTokens;
+  square?: JSAvatarTokens;
+
+  /**
+   * Token for inactive value of `active` prop
+   */
+  inactive?: JSAvatarTokens;
+
+  /**
+   * The width of the avatar.
+   */
+  width?: number;
+
+  /**
+   * Avatar sizes:
+   */
+  size20?: JSAvatarTokens;
```

Comment:
Would it be helpful to rename the existing Avatar to something like "NativeAvatar", so that we can rename JSAvatar to just Avatar? That's something we can coordinate with our client team to do if needed.

Reply:
I think yes. I'll create a task for that